modules/javafx.graphics/src/main/java/com/sun/glass/utils/NativeLibLoader.java

```diff
@@ -150,6 +161,29 @@ private static void loadLibraryInternal(String libraryName) {
         }
     }
 
+   /**
+    * If there is a library with the platform-correct name at the
+    * root of the resources in this jar, use that.
+    */
+    private static boolean loadLibraryFromResource(String libraryName, Class caller) {
+        try {
+            String reallib = "/"+libPrefix+libraryName+libSuffix;
+            InputStream is = caller.getResourceAsStream(reallib);
```

Comment:
This looks like a good approach. It might require a few more internal qualified exports from calling modules to javafx.graphics, but that won't be a problem.